src/Raven.Server/Documents/Queries/AST/PatternMatchElementExpression.cs

```diff
@@ -137,7 +137,7 @@ public RecursiveOptions GetOptions(QueryMetadata queryMetadata, BlittableJsonRea
                 case 0:
                     break;
                 case 1:
-                    value = QueryBuilder.GetValue(queryMetadata.Query, queryMetadata, queryParameters, Options[0]);
+                    value = LuceneQueryBuilder.GetValue(queryMetadata.Query, queryMetadata, queryParameters, Options[0]);
```

Comment:
Please move it to the query builder helper class. It's not Lucene specific. Here and in other places not specific to Lucene as well.